File: packages/cpp/src/leetcode/by-problem/0344-reverse-string.cpp
/*
    344 - Reverse String
    Two pointers | Middle Convergence
    Time: O(n) | Space: O(1)
    Do not return anything, modify s in-place instead.
*/

#include <vector>
#include <utility>

class Solution
{
public:
    void reverseString(std::vector<char> &s)
    {
        int l = 0;
        int r = s.size() - 1;

        while (l < r)
        {
            std::swap(s[l], s[r]);

            l++;
            r--;
        }
    }
};

File: packages/javascript/src/leetcode/by-problem/0344-reverse-string.js
/**
 * 344 - Reverse String
 * Two pointers | Middle Convergence
 * Time: O(n) | Space: O(1)
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
const reverseString = function (s) {
  let l = 0;
  let r = s.length - 1;

  while (l < r) {
    let left = s[l];
    let right = s[r];
    s[l] = right;
    s[r] = left;

    l++;
    r--;
  }
};
